code/modules/research/nanites/nanite_programmer.dm

```diff
@@ -139,5 +140,7 @@
 /obj/machinery/nanite_programmer/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, list/message_mods = list())
 	. = ..()
 	var/static/regex/when = regex("(?:^\\W*when|when\\W*$)", "i") //starts or ends with when
-	if(findtext(raw_message, when) && !istype(speaker, /obj/machinery/nanite_programmer))
+	if(findtext(raw_message, when) && !istype(speaker, /obj/machinery/nanite_programmer) && cooldown + 5 SECONDS < world.time)
 		say("When you code it!!")
+		cooldown = world.time
+		
```

Comment:
```suggestion

```

Reply:
I recommend you use existing macros for cooldowns. See the file that contains COOLDOWN_DECLARE

Reply:
Fixed! I avoided using that macro because it looked qutie intimidating on first glance. Also Mark, suffer.

Reply:
Lizard assassins sent directly to your house, Mark mustn't suffer